make/helm.mk

```diff
@@ -1,33 +1,46 @@
 ##@ Helm Charts
 
+# Chart name
+CHART_NAME ?= dns-operator
+# Chart directory
+CHART_DIRECTORY ?= charts/$(CHART_NAME)
+
 .PHONY: helm-build
 helm-build: yq manifests kustomize operator-sdk ## Build the helm chart from kustomize manifests
 	# Replace the controller image (Should remain consistent with what `make bundle` does)
 	cd config/manager && $(KUSTOMIZE) edit set image controller=$(IMG)
 	# Build the helm chart templates from kustomize manifests
-	$(KUSTOMIZE) build config/helm > charts/dns-operator/templates/manifests.yaml
-	V="$(VERSION)" $(YQ) eval '.version = strenv(V)' -i charts/dns-operator/Chart.yaml
-	V="$(VERSION)" $(YQ) eval '.appVersion = strenv(V)' -i charts/dns-operator/Chart.yaml
+	$(KUSTOMIZE) build config/helm > $(CHART_DIRECTORY)/templates/manifests.yaml
+	V="$(VERSION)" $(YQ) eval '.version = strenv(V)' -i $(CHART_DIRECTORY)/Chart.yaml
+	V="$(VERSION)" $(YQ) eval '.appVersion = strenv(V)' -i $(CHART_DIRECTORY)/Chart.yaml
 
 .PHONY: helm-install
 helm-install: $(HELM) ## Install the helm chart
 	# Install the helm chart in the cluster
-	$(HELM) install dns-operator charts/dns-operator
+	$(HELM) install $(CHART_NAME) $(CHART_DIRECTORY)
 
 .PHONY: helm-uninstall
 helm-uninstall: $(HELM) ## Uninstall the helm chart
 	# Uninstall the helm chart from the cluster
-	$(HELM) uninstall dns-operator
+	$(HELM) uninstall $(CHART_NAME)
 
 .PHONY: helm-upgrade
 helm-upgrade: $(HELM) ## Upgrade the helm chart
 	# Upgrade the helm chart in the cluster
-	$(HELM) upgrade dns-operator charts/dns-operator
+	$(HELM) upgrade $(CHART_NAME) $(CHART_DIRECTORY)
+
+# GPG_KEY_UID: substring of the desired key's uid, the name or email
+GPG_KEY_UID ?= 'Kuadrant Development Team'
+# GPG_KEYRING_BASE64: the gpg keyring base64 encoded
+GPG_KEYRING_BASE64 ?= <KUADRANT_GPG_KEYRING_BASE64>
 
 .PHONY: helm-package
-helm-package: $(HELM) ## Package the helm chart
+helm-package: $(HELM) ## Package the helm chart and GPG sign it
+	# Store the key
+	mkdir -p .gpg
+	echo $(GPG_KEYRING_BASE64) | base64 -d > .gpg/kuadrantsecring.gpg  #storing base64 GPG key into keyring
```

Comment:
writing the key to the disk does not seem to be a good idea. Maybe there is no other way, so let it be. As I see, the `helm-package` makefile target should assume that the key is already in the default keyring (or keystore) of the system, so it should only run `helm package --sign --key whatever`, without specifying the keyring. The only thing that matters is the key name that should exist. So, manually, the key should already been imported, and for GHA, it would be required a previous step for importing the key, something like:

```yaml
- name: Configure GPG Key
  run: |
    echo -n "$GPG_SIGNING_KEY" | base64 --decode | gpg --import
  env:
    GPG_SIGNING_KEY: ${{ secrets.HELM_CHARTS_SIGNING_KEY }}
```

wdyt?

Reply:
yup, it makes sense, let me do some changes addressing this, thanks for the suggestion!